### PR TITLE
Add docs for the JS/TS SDK

### PR DESCRIPTION
**Pull request by karthik2804**
Signed-off-by: karthik Ganeshram <karthik.ganeshram@fermyon.com>

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://bartholomew.fermyon.dev/quickstart)
- [x] Has run `npm run build-index`
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [ ] Have tested with `npm run test` and resolved all errors

### CODE REVIEW

**Comment by karthik2804** [comment]
There is one broken link due to the "Edit on Github"  button pointing to the page in the `main` branch.

File: content/spin/javascript-components.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
+title = "Building Spin Components in JavaScript"
+template = "spin_main"
+date = "2022-03-14T00:22:56Z"
+enable_shortcodes = true
+---
+
+With Javascript being a very popular language, Spin provides support for building components with it using the experimental SDK. The development of the Javascript SDK is continually being worked on to improve user experience and add features.
```
Comment (dicej) [inline content/spin/javascript-components.md:7]: I think the "official" spelling is "JavaScript" (with a capital "S"), so we should adhere to that.

File: content/spin/javascript-components.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+A new JS/TS component can be created using the following command:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin new http-ts hello-world --accept-defaults
+```
+
+This creates a directory of the following structure"
```
Comment (dicej) [inline content/spin/javascript-components.md:61]: Stray quotation mark?

File: content/spin/javascript-components.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+│   └── index.ts
+├── tsconfig.json
+└── webpack.config.js
+```
+
+The source for the component is present in `src/index.ts`. `Webpack` is used to bundle the component into a single `js` file which will then be compiled to a `wasm` module using the `js2wasm` plugin.
+
+## Building and Running the Template
+
+First, the dependencies for the template need to be installed and then bundled into a single JavaScript file using the following commands:
```
Comment (tpmccallum) [inline content/spin/javascript-components.md:81]: Perhaps include a suggestion to `cd hello-world` if you think it is necessary.

**Review by tpmccallum — APPROVED** [review]
Can we please remove the `package-lock.json` file.
If so, LGTM (Thanks for pairing on this with me Karthik, the examples in the blog work well on my machine).

File: content/spin/javascript-components.md (hunk 3 above)
Comment (karthik2804) [reply content/spin/javascript-components.md:81]: Good spot! I will add it.

File: content/spin/javascript-components.md [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+---
+
+With JavaScript being a very popular language, Spin provides support for building components with it using the experimental SDK. The development of the Javascript SDK is continually being worked on to improve user experience and add features. 
+
+> This guide assumes you are familiar with the Javascript programming language,
+> but if you are just getting started, be sure to check [the MDN guide](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide).
+
+> All examples from this page can be found in [the Spin repository on GitHub](https://github.com/fermyon/spin-js-sdk/tree/main/examples).
+
+In order to compile JavaScript programs to Spin components, you also need to install a spin plugin `js2wasm` using the following command:
```
Comment (radu-matei) [inline content/spin/javascript-components.md:14]: Capitalization for "Spin".

File: content/spin/javascript-components.md [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+
+<!-- @selectiveCpy -->
+
+```bash
+$ spin plugin update
```
Comment (radu-matei) [inline content/spin/javascript-components.md:19]: Is this command always needed?

File: content/spin/javascript-components.md [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+├── package-lock.json
+├── README.md
+├── spin.toml
+├── src
+│   └── index.ts
+├── tsconfig.json
+└── webpack.config.js
+```
+
+The source for the component is present in `src/index.ts`. `Webpack` is used to bundle the component into a single `js` file which will then be compiled to a `wasm` module using the `js2wasm` plugin.
```
[the remark this answers is not in the document]
Comment (radu-matei) [reply content/spin/javascript-components.md:77]: Also: "into a single JavaScript file which will be then compiled to a Wasm module" — not sure about the code block for `js` and `wasm`.

(alternatively, if you want to use the file extension, that would make sense — so `.js`, `.wasm`)

File: content/spin/javascript-components.md [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+## Building and Running the Template
+
+First, the dependencies for the template need to be installed and then bundled into a single JavaScript file using the following commands:
+
+<!-- @selectiveCpy -->
+
+```bash
+$ cd hello-world
+$ npm install
+$ spin build
```
Comment (radu-matei) [inline content/spin/javascript-components.md:88]: I think the blog post uses `nom run build`, which calls `spin build` — I don't have a preference towards one or the other, but I think we should be consistent here?

File: content/spin/javascript-components.md (hunk 5 above)
Comment (karthik2804) [reply content/spin/javascript-components.md:19]: This is currently needed if someone has already installed some plugin. In most cases no but without it there is a possibility that the next command will throw an error saying manifest for js2wasm not found.

File: content/spin/javascript-components.md [hunk 8]
```diff
@@ -0,0 +1,281 @@
+title = "Building Spin Components in JavaScript"
+template = "spin_main"
+date = "2022-03-14T00:22:56Z"
+enable_shortcodes = true
+---
+
+With JavaScript being a very popular language, Spin provides support for building components with it using the experimental SDK. The development of the Javascript SDK is continually being worked on to improve user experience and add features.
```
[the remark this answers is not in the document]
Comment (karthik2804) [reply content/spin/javascript-components.md:7]: Would that require rephrasing of the content?

File: content/spin/javascript-components.md [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+}
+```
+
+This HTTP component demonstrates fetching a value from Redis by key, setting a key with a value, and publishing a message to a Redis channel. The component is triggered by an HTTP request served on the route configured in the `spin.toml`:
+
+> When using Redis databases hosted on the internet (i.e) not on localhost, the `redisAddress` must be of the format "redis://\<USERNAME\>:\<PASSWORD\>@\<REDIS_URL\>" (e.g) `redis://myUsername:myPassword@redis-database.com`
+
+## Using External NPM Libraries
+
+> Not all the npm packages are guaranteed to work with the SDK as it is not fully compatible with the browser or nodejs. It implements only a subset of the API.
```
Comment (radu-matei) [inline content/spin/javascript-components.md:267]: NPM and Node.js are the right capitalizations here.

File: content/spin/javascript-components.md [hunk 10]
```diff
@@ -0,0 +1,281 @@
+title = "Building Spin Components in JavaScript"
+template = "spin_main"
+date = "2022-03-14T00:22:56Z"
+enable_shortcodes = true
+---
+
+With JavaScript being a very popular language, Spin provides support for building components with it using the experimental SDK. The development of the Javascript SDK is continually being worked on to improve user experience and add features. 
+
+> This guide assumes you are familiar with the Javascript programming language,
```
[the remark this answers is not in the document]
Comment (radu-matei) [reply content/spin/javascript-components.md:9]: Is this resolved?
Or something you will push soon?

File: content/spin/javascript-components.md [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+### Suggested Libraries for Common Tasks
+
+These are some of the suggested libraries that have been tested and confired to work with the SDK for common tasks.
+
+{{ details "Parsing formdata" "- [parse-multipart-data](https://www.npmjs.com/package/parse-multipart-data)" }} 
+
+## Caveats
+
+- All `spinSdk` related functions and methods can be called only inside the `handleRequest` function. This includes the usage of `fetch`. Any attempts to use it outside the function will lead to an error. This is due to Wizer using only Wasmtime to execute the script at build time, which does not include any Spin SDK support.
+- Only a subset of the browser and `node.js` APIs are implemented.
```
Comment (radu-matei) [inline content/spin/javascript-components.md:280]: Node.js*

**Review by radu-matei — APPROVED** [review]
This looks great, thank you!

**Pull request merged**